appcelerator/titanium_mobile
# [TIMOB-25173] Android: Modified layouts to behave more like iOS.

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-25173

Also resolves the following JIRA cases...
- [TIMOB-12996](https://jira.appcelerator.org/browse/TIMOB-12996)
- [TIMOB-15097](https://jira.appcelerator.org/browse/TIMOB-15097)
- [TIMOB-17628](https://jira.appcelerator.org/browse/TIMOB-17628)
- [TIMOB-17984](https://jira.appcelerator.org/browse/TIMOB-17984)
- [TIMOB-19536](https://jira.appcelerator.org/browse/TIMOB-19536)
- [TIMOB-19598](https://jira.appcelerator.org/browse/TIMOB-19598)
- [TIMOB-19814](https://jira.appcelerator.org/browse/TIMOB-19814)

**Summary:**
- Changed Android to resolve Size/Fill layout conflicts like iOS and Windows.
  * Note: Only an issue if parent uses Ti.UI.SIZE and child uses Ti.UI.FILL.
  * Android used to shrink the parent to fit the min size allowed by the child.
  * Now increases the size of the child to fill the parent's parent, like iOS/Windows.
- Fixed horizontal layouts where views using FILL width would wrongly wrap to the next row.
  * Only an issue if another view is already in the row. (Resulting FILL width was too big.)
- Fixed layouts to size itself using view's top/left/bottom/right/center properties if width or height property was not set.
  * Only applies if at least 2 coordinate properties were provided.
  * Now matches iOS' behavior.
- Constrained child view width to parent's max width like iOS when using wrapping horizontal views.

**Tests:**
- Run the following 4 test apps attached to [TIMOB-25174](https://jira.appcelerator.org/browse/TIMOB-25173) and verify that the results match the "good" screenshots.
  * `HorizontalFillTest.js`
  * `HorizontalPinningTest.js`
  * `ListViewFillTest.js`
  * `SizeFillConflictTest.js`
  * `VerticalFillTest.js`
- Run the main test code in [TIMOB-19598](https://jira.appcelerator.org/browse/TIMOB-19598) and verify that the horizontal layout matches the attached iOS screenshot.
- Run test code in [TIMOB-15097](https://jira.appcelerator.org/browse/TIMOB-15097) and verify that result matches iOS screenshot.
- Run main test code in [TIMOB-17628](https://jira.appcelerator.org/browse/TIMOB-17628) and verify ListView is displayed fullscreen.
- Run test code in PR #9073 .
- Run test code in PR #9123 .
